src/DynamoCoreWpf/Interfaces/IWatchHandler.cs

```diff
@@ -51,17 +52,32 @@ public DefaultWatchHandler(IPreferences preferences)
 
         private WatchViewModel ProcessThing(object value, ProtoCore.RuntimeCore runtimeCore, string tag, bool showRawData, WatchHandlerCallback callback)
         {
-            WatchViewModel node;
+            if (value is DesignScript.Builtin.Dictionary)
+            {
+                var dict = value as DesignScript.Builtin.Dictionary;
+                var keys = dict.Keys;
+                var values = dict.Values;
+
+                var node = new WatchViewModel(keys.Any() ? WatchViewModel.DICTIONARY : WatchViewModel.EMPTY_DICTIONARY, tag, RequestSelectGeometry, true);
+
+                foreach (var e in keys.Zip(values, (key, val) => new { key, val }))
+                {
+                    node.Children.Add(ProcessThing(e.val, runtimeCore, tag + ":" + e.key.ToString(), showRawData, callback));
+                }
 
-            if (value is IEnumerable)
+                return node;
+            }
+            else if (!(value is string) && value is IEnumerable)
```

Comment:
`string` is `IEnumerable`. Without adding this check, preview bubbles can represent strings as a `List` of integers.

Reply:
`String` is `IEnumerable`??

Reply:
```public sealed class String : IComparable, ICloneable, IConvertible, IEnumerable, IComparable<string>, IEnumerable<char>, IEquatable<string>``` from https://msdn.microsoft.com/en-us/library/system.string(v=vs.110).aspx